File: pkg/pwrgrep/testdata/fixtures/go/entry-points.go
package fixture

import (
	"net/http"
	"testing"
)

// ruleid: go-entry-points
func main() {
	// ruleid: go-entry-points
	http.HandleFunc("/hello", hello)
	// ruleid: go-entry-points
	r.GET("/users/:id", getUser)
	// ok: go-entry-points
	cache.Get("/not-a-route", 1)
}

// ruleid: go-entry-points
func init() {}

// ruleid: go-entry-points
func TestHello(t *testing.T) {}

// ok: go-entry-points
func hello(w http.ResponseWriter, req *http.Request) {}

File: pkg/pwrgrep/testdata/fixtures/python/entry-points.py
import click
from flask import Flask

app = Flask(__name__)


@app.route("/hello")
# ruleid: python-entry-points
def hello():
    return "hi"


@app.get("/items/{id}")
# ruleid: python-entry-points
async def item(id):
    pass


@click.command()
# ruleid: python-entry-points
def cli():
    pass


# ruleid: python-entry-points
def test_hello():
    assert hello() == "hi"


# ok: python-entry-points
def helper():
    pass


# ruleid: python-entry-points
if __name__ == "__main__":
    cli()
